docs/REDUCERS.md

```diff
@@ -0,0 +1,347 @@
+```
+to allow us to later define when we want to receive an **action** for our `Index`-page and later combine them to create a shared type for all the **actions** in our application.
+
+---
+
+Next we define our [Epics](https://redux-observable.js.org/docs/basics/Epics.html)
+```typescript
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/delay';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/mapTo';
+import { Epic, combineEpics, ActionsObservable } from 'redux-observable';
+
+const saveTodoEpic: Epic<IndexActions, undefined> = (action$: ActionsObservable<IndexActions>): Observable<IndexActions> =>
+    action$.ofType(SAVE_TODO)
+        .delay(1000)
+        .mapTo(saveTodoSuccess());
+
+const setDoneEpic: Epic<IndexActions, undefined> = (action$: ActionsObservable<IndexActions>): Observable<IndexActions> =>
+    action$.ofType(SET_DONE)
+        .delay(1000)
+        .map((action: SetDoneAction) => setDoneSuccess(action.payload));
+
+export const IndexEpics = combineEpics(saveTodoEpic, setDoneEpic);
+```
+which are [redux-observable's](https://redux-observable.js.org) way of handling side-effects in **Redux** (*like AJAX calls etc.*). At the end we combine all our **Epics** in this file to a single exportable **Epic** called `IndexEpics` (*so we only need to import one variable when we want access to these later*).
+> The importing part may look a little weird, but it's because [RxJS](http://reactivex.io/rxjs/) is a rather large library, we can either import everything using `import * as RxJS from 'rxjs'` or import only the parts we need as shown above, which will allow any proper [minifier](https://developers.google.com/speed/docs/insights/MinifyResources) like [UglifyJS](https://github.com/mishoo/UglifyJS) include only the needed parts from **RxJS**
+
+The first line
+```typescript
+const saveTodoEpic: Epic<IndexActions, undefined> = (action$: ActionsObservable<IndexActions>): Observable<IndexActions> =>
+```
+defines an **Epic** which takes in as the first type argument the `type` for the `Actions` the epic takes in (*and returns*), in this case `IndexActions` which we defined earlier, and as the second argument the type of the **State** it takes in (*which isn't needed this time, so undefined will do*). An **Epic** is a function that takes in a [stream](https://en.wikipedia.org/wiki/Stream_(computing)) (*in this case of the type `ActionsObservable`*) which includes items of the `type` given as the first type argument and returns another stream (*in this case an [`Observable`](http://reactivex.io/documentation/observable.html), which `ActionsObservable` is based on*) which includes items of the same `type` as the input stream.
+> In JavaScript the convention is to append a `$` to all variables names that are **streams**, to let the developer know that they are dealing with one
```

Comment:
https://cycle.js.org/basic-examples.html#basic-examples-increment-a-counter-what-is-the-convention <- it also allows the same kind of naming logic as e.g. plurals for arrays :) Think of the example `name$.map(name => doSomething(name))` versus `name.map(n => doSomething(n))`